GameMod/MPInternet.cs

```diff
@@ -48,6 +48,38 @@ public static IPAddress FindPasswordAddress(string password, out string msg)
             }
             return null;
         }
+        public static IPAddress[] ResolveIPAddress(string name)
+        {
+            bool foundV4 = false;
+            IPAddress[] addrs = Dns.GetHostAddresses(name); // async is better but would need moving bottom of NetworkMatch.SwitchToLobbyMenu to NetSystemDoTick
+            // the game doesn't like IPv6,
+            // so for now, try to find a V4 address if there is any
+            // The caller of this function simply takes the first element of the array,
+            // so we can sort the V4 to the first place if there is one
+            if (addrs != null && addrs.Length > 0) {
```

Comment:
Is there a reason why you don't use Linq to filter what you want, like
`var ipv4addrs = Dns.GetHostAddresses(name).Where(a => a.AddressFamily == AddressFamily.InterNetwork).ToArray()`